src/pkg/user/service/service.go

```diff
@@ -22,77 +28,247 @@ type Service struct {
 	Proc  *processing.Proc
 }
 
+type TokenMetadata struct {
+	UUID   string
+	UserID string
+}
+
+type TokenDetails struct {
+	AccessToken  string
+	RefreshToken string
+	AccessUUID   string
+	RefreshUUID  string
+	AtExpires    int64
+	RtExpires    int64
+}
+
 const (
-	N            = 1048576
-	r            = 8
-	p            = 1
-	keyLen       = 32
-	saltLenBytes = 16
+	N      = 1048576
+	r      = 8
+	p      = 1
+	keyLen = 32
 )
 
 func NewUserService(cfg *config.Config, infra *infrastructure.Infra) *Service {
-	proc := processing.New(
+	process := processing.New(
 		infra.LocalDB,
 		infra.EthClient,
 		infra.FilecoinClient,
 		infra.IpfsClient,
 		cfg.Storage.Localfile.Path,
 	)
 
-	proc.Start()
+	process.Start()
 
 	return &Service{
 		Infra: infra,
-		Proc:  proc,
+		Proc:  process,
 	}
 }
 
 func (s *Service) Register(ctx context.Context, procRequest *proc.Request, user *model.UserCreateRequest) (err error) {
-	_, userPrivateKey, err := s.Infra.Keystore.Get(user.UserID)
+	ehrSystemID := ctx.(*gin.Context).GetString("ehrSystemID")
+	address, pwdHash, err := s.getUserAddressAndHash(ehrSystemID, user.UserID, user.Password)
```

Comment:
getUserAddressAndHash - Hash of what?
needs to change the function name or split it.

Reply:
\+